compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -4593,8 +4593,14 @@ private STNode createMissingSimpleObjectField(STNode metadata, List<STNode> qual
 
         simpleNameRef = modifyNodeWithInvalidTokenList(qualifiers, simpleNameRef);
 
-        return STNodeFactory.createObjectFieldNode(metadata, emptyNode, objectFieldQualNodeList, simpleNameRef,
-                identifier, emptyNode, emptyNode, semicolon);
+        STNode objectField = STNodeFactory.createObjectFieldNode(metadata, emptyNode, objectFieldQualNodeList,
+                simpleNameRef, identifier, emptyNode, emptyNode, semicolon);
+
+        if (metadata != null) {
+            objectField = SyntaxErrors.addDiagnostic(objectField,
```

Comment:
```suggestion
            return SyntaxErrors.addDiagnostic(objectField,
```

Reply:
@gimantha This is not an early exit condition. In this case single return at L4603 looks clean to me. Multiple return is a bit confusing. Also returning with `SyntaxErrors.addDiagnostic()` method call is confusing. WDYT?

Reply:
Yes, I used this way mainly because of the name. Since `SyntaxErrors.addDiagnostic()` doesn't imply any return value, we usually don't  return with it